x-pack/test/security_solution_cypress/cypress/tasks/api_calls/alerts.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { rootRequest } from '../common';
+
+export const DEFAULT_ALERTS_INDEX_PATTERN = '.alerts-security.alerts-*';
+
+interface FetchRuleAlertsParams {
+  ruleId: string;
+  fields: string[];
+  size?: number;
+}
+
+interface AlertFields {
+  fields: Record<string, unknown>;
+}
+
+interface AlertsResponse {
+  hits: {
+    total: {
+      value: number;
+    };
+    hits: AlertFields[];
+  };
+}
+
+export function fetchRuleAlerts({
+  ruleId,
+  fields,
+  size,
+}: FetchRuleAlertsParams): Cypress.Chainable<Cypress.Response<AlertsResponse>> {
+  return rootRequest<AlertsResponse>({
+    method: 'GET',
+    url: `${Cypress.env('ELASTICSEARCH_URL')}/${DEFAULT_ALERTS_INDEX_PATTERN}/_search`,
+    body: {
+      query: {
+        bool: {
+          should: [
+            {
+              term: {
+                'kibana.alert.rule.rule_id': ruleId,
+              },
+            },
+            {
+              term: {
+                'kibana.alert.rule.uuid': ruleId,
+              },
+            },
+          ],
```

Comment:
I think we should only check the `kibana.alert.rule.uuid` field, which corresponds to the SO id of the rule. The `kibana.alert.rule.rule_id` represents the `rule.rule_id` field of type `RuleSignatureId`, and it's a completely different id.

I'm wondering why in this test they are equal. Let's figure this out to make sure the test doesn't contain a bug.

Reply:
The problem with `kibana.alert.rule.uuid` it's unknown until a rule is created. To pass it to the test extra code is required making it more bulky. `rule_id` on the other side is known beforehand so a constant can be used. Of course these two ids are usually different. I just made it possible to fetch rule's alert by whatever its ids and use `rule_id` to for sake of simplicity. Do we plan to discontinue `rule_id` so SO `id` is preferrable?

Reply:
Sorry, I overlooked the `should` clause. Let's add JSDoc comments to this function explaining that it can fetch alerts either by `RuleObjectId` or by `RuleSignatureId`.